shared/actions/chat2/index.tsx

```diff
@@ -2535,19 +2535,27 @@ const navigateToThread = (action: Chat2Gen.NavigateToThreadPayload) => {
     return false
   }
 
+  const currentPath = Router2Constants.getVisiblePath()
+  const modalPath = Router2Constants.getModalStack()
+  const mainPath = Router2Constants.getMainStack()
+
+  const modalClearAction = modalPath.length > 0 ? [RouteTreeGen.createClearModals()] : []
+  const tabSwitchAction =
+    mainPath[1]?.routeName !== Tabs.chatTab ? [RouteTreeGen.createSwitchTab({tab: Tabs.chatTab})] : []
```

Comment:
@chrisnojima Do you think this kind of logic should be included in `RouteTreeGEn.createSwitchTab()`?

Reply:
possibly but i want to change this minimally now. nav5 is in progress and we'll make larger changes then